<?xml version="1.0"?>
<launch>
    <arg name="video_width" default="1280" />
    <arg name="video_height" default="720" />

    <arg name="usb_cam" default="0" />
    <arg name="gpuid" default="0" />

    <arg name="model" default="mobilenet_thin" />
    <arg name="resolution" default="432x368" />

    <node name="pose_estimator" pkg="tfpose_ros" type="broadcaster_ros.py" output="screen" required="true">
        <env name="CUDA_VISIBLE_DEVICES" value="$(arg gpuid)" />
        <param name="camera" value="/camera/image_raw" />
        <param name="model" value="$(arg model)" />
        <param name="resolution" value="$(arg resolution)" />
    </node>

    <node name="pose_estimator_vis" pkg="tfpose_ros" type="visualization.py" output="screen" required="true">
        <param name="camera" value="/camera/image_raw" />
        <param name="pose" value="/pose_estimator/pose" />
    </node>

    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen" required="true">
        <remap from="image" to="/pose_estimator_vis/output"/>
        <param name="autosize" value="true" />
    </node>

    <include file="$(find video_stream_opencv)/launch/camera.launch">
        <arg name="video_stream_provider" value="$(arg usb_cam)" />
        <arg name="width" value="$(arg video_width)"/>
        <arg name="height" value="$(arg video_height)"/>
    </include>

</launch>